.github/workflows/ubuntu.yml

```diff
@@ -3,28 +3,27 @@ name: ubuntu
 
 on: [push, pull_request]
 
-## Unfortunately, we didn't manage to use something like "variables"
-## globally. Github actions can set environment variables globally,
-## but they can only be single values, not lists, and ideally we would
-## like to have something like COMPILER_VERSIONS = [gcc, ..].
-## Now, whenever we change versions, we have to remember to do this
-## *everywhere* in this file.
-
 jobs:
-  compile: # identifier of the job. Jobs run in parallel unless specified otherwise.
+  compile: # Identifier of the job. Jobs run in parallel unless specified otherwise.
     name: compile
     runs-on: ${{ matrix.ubuntu-version }}
     strategy:
       matrix:
         ubuntu-version: [ubuntu-18.04, ubuntu-20.04]
-        compiler-version: [gcc, gcc-10, clang, clang-11]
+        compiler-version:
+          - {cc: gcc, cxx: g++}
+          - {cc: gcc-10, cxx: g++-10}
+          - {cc: clang, cxx: clang++}
+          - {cc: clang-11, cxx: clang++-11}
         python-version: [3.6]
+        # Unfortunately, we couldn't figure out if there is a way to
+        # name the compiler versions so that we don't have to copy them here.
```

Comment:
I just noticed that the debug builds are missing in this workflow.

Reply:
No, for every matrix strategy, build for debug and release. See below.

Reply:
My bad, you're right. But wouldn't it make sense to add the "release" vs. "debug" split to the build matrix as for Windows and macOS?

Reply:
Initially, we split the compilation of release and debug, because we could parallize this well(16min -> 2x8min). The setup for the compilation is negligible. The uploading and downloading of the artifacts takes some time, but is still a speedup (+2-4min)
For Ubuntu, we also install the LP solvers. This takes quite some time (+8 minutes), thus, our impression is that the speed up from parallelization is not making up for the additional wasted resources.

(In issue1005 we add LP support to windows and I removed the parallelization there too)

Reply:
Ah, I see, thanks!